### ISSUE TITLE
Routine Maintenance very slow

### ISSUE THREAD

**Issue by aterriblegamer** [jlesage/docker-crashplan-pro#232]
Pruning just took 5.7 days
Compacting (deep) 5.9 days to go

Is there some setting I should be looking at to improve this performance?

**Comment by jlesage** [comment]
With the size of you backup, I'm not surprised: CP is known to be very slow...

There are settings to set the frequency of shallow and deep maintenances.  Maybe this is something you can increase... ?

**Comment by madfusker** [comment]
I see similar times for my 4.1 TB archive on a QNAP TS-877. About 7 days to do both those operations.  The 877 is dual AMD Ryzen 7 1700 Eight-Core Processor, so 16 cores.  A beast!  I believe the slowness here is fully on the Crashplan side of things.  My guess is they have lower priority VMs doing this work and thus takes forever.

**Comment by johnripper1** [comment]
You cant speed up the maintenance tasks on the client side.

Maybe the CP support can, but nothing jlesage, you or the community can do.

Can be closed IMHO

**Comment by johnripper1** [comment]
BTW, i have a 15TB backup set and the job is running 14 days or more.

**Comment by madfusker** [comment]
Yeah, probably can close this one.  This does however seem pretty excessive on CrashPlan's side...  If you ask for any support they will want your setup details, and when they find out it is docker or a NAS, they will likely say it's not supported.  Frustrating!

**Comment by jlesage** [comment]
> If you ask for any support they will want your setup details, and when they find out it is docker or a NAS, they will likely say it's not supported.

Does this happen to you recently ?  If yes, I would love to know which information they use to conclude that CrashPlan is used in a Docker or NAS...

**Comment by jvarrone** [comment]
"...which information they use to conclude that CrashPlan is used in a Docker or NAS..."
Only if you tell them. My issue was and still is,  the Docker setup not seeing the /volume1 so I had to reveal it to them.

**Comment by madfusker** [comment]
> Does this happen to you recently ? If yes, I would love to know which information they use to conclude that CrashPlan is used in a Docker or NAS...

Not in a year or so when I had to go to support for a cloud connect issue.  I had my backup unit named "NAS" so they hit me up right away.  I renamed it to Ubuntu after that.  I wasn't sure if the app reported up any arch or hardware ID?  When they ask for OS, what should we report?  Call it Ubuntu?  I am not sure what all they can see on their side, and maybe it looks like debian or something to them if it shows kernel version, etc.

@jvarrone I haven't had any issues with volumes, but mine are quite simple with /share/blah

**Comment by jlesage** [comment]
I've taken some steps to make sure the app doesn't report the real host's kernel name/version.  So the info reported by the app should be the similar to a real Ubuntu/debian distro.
Renaming your device name and backup set to something that doesn't reveal your environment is also a good idea.

**Comment by madfusker** [comment]
I poked the bear and got the usual response.  I've pasted it below, read top down.  I don't know how it can take a week to scan 4.9TB of data, but.....   

BTW - I still have 16.5 hours to go, and here it is May 22 at 1435.  

-----------------------------------------
Hello,

 My archive is doing routine maintenance pruning versions deeps right now, and I've noticed over the last year or so, the time it takes to complete is getting extremely long while the size of the archive is still about the same size. Maintenance started on May 17th at 1641, and as of today May 20 at 2013, it still shows 59% complete and another 2.1 days to go! This means for a 4.9TB archive, I need to wait 5-6 days (up to 144 hours) just for it to do simple routine maintenance? This also means Crashplan is not backing up my daily data during this time and leaves my data unprotected and at risk for nearly a full week!

 Why is routine maintenance taking so long? Is my archive on a VM that is so de-prioritized that it takes a week to scan though less than 5TB of data? How can I get my data on a higher priority VM that doesn't put my daily data at risk from waiting for maintenance to finish? I would expect this to take 3-4 hours at best, not 144 hours! This is a business plan, so if I were a business backing up 100TB of data, you'd make my business wait 100 days to do maintenance before I could backup data again? (5TB takes 5 days, so 100TB would be 100 days).  A business plan user cannot have data loss exposure risk for this long waiting to backup.  Can you offer any help to speed this up?

Linux
Crashplan v 8.0.0.778
Raid 5

Thank you.
------------------------------------------------------------------------------------
Hello,

We do not prioritize which backup archives are maintained first; each archive is added to the maintenance queue for the server in the order that automatic maintenance on them began. We do not offer an option to "prioritize VMs" as you're describing. If you want to decrease how long it takes for your backup archive to complete maintenance, you would need to decrease the size of the data you're storing in the backup archive. A majority of CrashPlan users have a backup archive less than 100GB in size for each device, and see maintenance take at most 1-2 hours - when you're backing up a large amount of data, you will see the length of time it takes to maintain that data increase. I would agree that it wouldn't be viable to store 100TB of data from a single device with our service - aside from how long maintenance would take, the average upload speed is 10-30GB a day, so uploading that much initially would take at best 8-9 years to complete.

CrashPlan is designed to be a data redundancy service, not an archival service. One reason why you should keep an original copy of your files on your computer is due to how long archive maintenance may take if you're backing up a huge amount of data with us. Your data is extremely important to us. So important, in fact, that we strongly believe you shouldn't trust just a single provider or destination, even our cloud. We give you the ability to back up to multiple destinations. We always recommend that you back up to both local and offsite destinations.

If there is anything further I can clarify for you, please let me know, and I'll get back to you as soon as possible

Regards,
Jonathan K
Crashplan for Small Business
Technical Support Agent
Contact us via live chat support

**Comment by jvarrone** [comment]
"We give you the ability to back up to multiple destinations." Since when? How?

**Comment by madfusker** [comment]
Well, you can also add a "local" destination in your backup sets.  This would put crashplan files on the local NAS in which you could one-way sync it out to another NAS somewhere I suppose.  I haven't done that as it seems a bit stupid.  Instead I have been looking at QNAP HBS 3 to sync to another remote NAS as my secondary backup.  It's a bit more failsafe than trusting crashplan on two destinations.  If Crashplan screws up my archive, you can bet it will be on both of them.

**Comment by jvarrone** [comment]
I looked at the backup sets and didn't see an option to backup locally.
I'm with you though, I'm going to get a single drive Synology, back up everything to it and store it at my friend's house. Synology has a utility to back itself up over WAN, incrementally.

**Comment by madfusker** [comment]
You have to go into the backup set and then you will see it.

![Capture](https://user-images.githubusercontent.com/39440750/82714547-d13f6480-9c54-11ea-8250-70f1a1503960.JPG)

**Comment by madfusker** [comment]
> 
> 
> I looked at the backup sets and didn't see an option to backup locally.
> I'm with you though, I'm going to get a single drive Synology, back up everything to it and store it at my friend's house. Synology has a utility to back itself up over WAN, incrementally.

Just buy the drive and give it to your friend.  If you have a slot for your friend offer that too in exchange.  If Synology works like QNAP, the data is all encrypted so no problem storing the hash on someone else's device.

**Comment by madfusker** [comment]
See updated pics post above on how to get there.
